### ISSUE TITLE
Repository got banned from mybinder.org

### ISSUE THREAD

**Issue by betatim** [jupyterlab/jupyterlab-demo#74]
Hello from mybinder.org,

we had a bit of a rough evening since you merged #73. We aren't quite sure why this caused us problems or what is wrong with the build for this repository. However we observed very high CPU load, and eventually noticed that lots of ipython demo pods were running but none from jupyterlab. The first time our CPU usage went through the roof was exactly when you merged #73. So for the moment we banned this repository. After the first few minutes it seems like this resolved our problem.

You can try and build the repository locally with `repo2docker` or making a fork which should not be banned.

You can find us on gitter: https://gitter.im/jupyterhub/binder or we discuss in this issue

**Comment by betatim** [comment]
Thanks for this :) We discovered a problem in how we handle lots of people trying to build the same repository that is taking a very long time to build. At least this seems to be one of the problems that we ran into tonight. So we have some homework to do.

One thing to consider as this is one of the highest traffic binders we have is to integrate something like this [continuous build setup](https://github.com/binder-examples/continuous-build/) into the CI for this repository. It isn't 100% clear if this would have prevented tonight's adventure as reproducing the long build time is proving tricky. Another thing to ponder is if we want to keep linking to `master` from http://jupyter.org/try or only link to known-to-be-good tags from there. This would have definitely prevented this but it introduces a manual step.
